test/integration-test/src/tests/tcx.rs

```diff
@@ -0,0 +1,185 @@
+    let mut program3 = EbpfLoader::new()
+        .set_global("ORDER", &3, true)
+        .load(crate::TCX)
+        .unwrap();
+
+    let logs0: Arc<Mutex<Vec<CapturedLog>>> = Arc::new(Mutex::new(Vec::new()));
+    setup_logs(&mut program0, &logs0);
+
+    let logs1 = Arc::new(Mutex::new(Vec::new()));
+    setup_logs(&mut program1, &logs1);
+
+    let logs2 = Arc::new(Mutex::new(Vec::new()));
+    setup_logs(&mut program2, &logs2);
+
+    let logs3 = Arc::new(Mutex::new(Vec::new()));
+    setup_logs(&mut program3, &logs3);
+
+    let prog0: &mut SchedClassifier = program0
+        .program_mut("tcx_order")
+        .unwrap()
+        .try_into()
+        .unwrap();
+    prog0.load().unwrap();
+
+    let prog1: &mut SchedClassifier = program1
+        .program_mut("tcx_order")
+        .unwrap()
+        .try_into()
+        .unwrap();
+    prog1.load().unwrap();
+
+    let prog2: &mut SchedClassifier = program2
+        .program_mut("tcx_order")
+        .unwrap()
+        .try_into()
+        .unwrap();
+    prog2.load().unwrap();
+
+    let prog3: &mut SchedClassifier = program3
+        .program_mut("tcx_order")
+        .unwrap()
+        .try_into()
+        .unwrap();
+    prog3.load().unwrap();
+
+    // Test LinkOrder::last() with correct expected_revision
+    let mut order: LinkOrder = LinkOrder::last();
+    order.set_expected_revision(1);
+    let options = TcAttachOptions::TcxOrder(order);
+    prog0
+        .attach_with_options("lo", TcAttachType::Ingress, options)
+        .unwrap();
+
+    // Test LinkOrder::after_program() with correct expected_revision
+    let mut order = LinkOrder::after_program(prog0).unwrap();
+    order.set_expected_revision(2);
+    let options = TcAttachOptions::TcxOrder(order);
+    let prog1_link_id = prog1
+        .attach_with_options("lo", TcAttachType::Ingress, options)
+        .unwrap();
+
+    let prog1_link = prog1.take_link(prog1_link_id).unwrap();
+
+    // Test incorrect expected_revision and expect an error
+    let mut order = LinkOrder::after_link(&prog1_link).unwrap();
+    order.set_expected_revision(7);
+    let options = TcAttachOptions::TcxOrder(order);
+    let result = prog2.attach_with_options("lo", TcAttachType::Ingress, options);
+    assert!(result.is_err());
+
+    // Test LinkOrder::after_link() again with expected_revision == 0 which
+    // means the expected_revision should be ignored.
+    let mut order = LinkOrder::after_link(&prog1_link).unwrap();
+    order.set_expected_revision(0);
+    let options = TcAttachOptions::TcxOrder(order);
+    prog2
+        .attach_with_options("lo", TcAttachType::Ingress, options)
+        .unwrap();
+
+    // Test LinkOrder::last() with no expected_revision
+    let options = TcAttachOptions::TcxOrder(LinkOrder::last());
+    prog3
+        .attach_with_options("lo", TcAttachType::Ingress, options)
+        .unwrap();
+
+    // Wait for system to stabilize
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
This test was failing for me locally because the log messages weren't in the expected order.  I added some debug code that did a BPF_PROG_QUERY syscall and confirmed that the the programs were actually installed in the right order.  I'm guessing that that there are some race conditions in the logging initialization, but I don't really know.  However, after adding a sleep here, it passed every time.  I'm proposing that we do a followup pr in which we use a new Aya API to call BPF_PROG_QUERY to verify the program order instead of using the logs.

Reply:
Can we rip out this nonsense of trying to observe ordering using userspace timestamps? Logging uses a percpu array, I'm not interested in debugging flakes in this test.

Reply:
Yes, I think we should.  I further explained the way I believe it was intended to work in a response to your comment on reviewable.  I also consolidated the logs to one log and had all bpf programs log to that same log, but it had the same problems where logs didn't happen in a deterministic order.  Are you okay with me removing the log-based order verification in this pr, and then adding a better way that uses a new API that gets the order info and also `expected_revision` via the `BPF_PROG_QUERY` syscall in a follow-on pr?

Reply:
Timestamp-based ordering tests have been removed.